//
// Copyright 2024 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package psi.spiral;

// for Raw and Ntt
message PolyMatrixProto {
  uint64 rows = 1;
  uint64 cols = 2;
  repeated uint64 data = 3;
}

message SpiralQueryProto {
  PolyMatrixProto ct = 1;
  bytes seed = 2;
}

message PublicKeysProto {
  repeated PolyMatrixProto v_packing = 1;

  repeated PolyMatrixProto v_expansion_left = 2;

  repeated PolyMatrixProto v_expansion_right = 3;

  repeated PolyMatrixProto v_conversion = 4;
}

message SpiralResponseProto {
  repeated PolyMatrixProto ct = 1;
}